navigator/runtime/src/main/java/com/freeletics/mad/navigator/NavEvent.kt

```diff
@@ -23,13 +24,24 @@ public sealed interface NavEvent {
     ) : NavEvent
 
     /**
-     * Navigates back to the given [popUpToDestinationId]. If [inclusive] is `true` the destination
+     * Navigates back to the given [popUpTo]. If [inclusive] is `true` the destination
      * itself will also be popped of the back stack. Then navigates to the given [route].
      */
     @VisibleForTesting(otherwise = PACKAGE_PRIVATE)
-    public data class NavigateBackAndThenToEvent(
+    public data class NavigateToOnTopOfEvent(
```

Comment:
Yes. The main reason that we have it as an extra event is that it is also a possibility with navcontroller (calling navigate with options that use popUpTo). It's a bit nicer because it doesn't rely on automatically merging multiple nav operations into one